### DISCUSSION TITLE
gsql-proxy endpoint causing additional delay in Query response

### DISCUSSION THREAD

**Discussion by Staking7pc** [subquery/subql#1039] [Frontier TestNet - General]
As part of querying  indexers all requests are routed to  https://gql-proxy.subquery.network and then redirected to the actual indexers  query node. 
For example query to our indexer **Brightlystake** is routed using the address https://gql-proxy.subquery.network/0xD3bA0e2067e2399Ca0Db5b5f03f416d1464Dd684?to=http%3A%2F%2Fsubquery-testnet.brightlystake.com%2F%2Fquery%2FQmYR8xQgAXuCXMPGPVxxR91L4VtKZsozCM7Qsa5oAbyaQ3 .

I believe that this is causing unintended delay in the response time. For a indexer present and running in Central Europe the below is the response time when queried from a server located in London.

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/Admin/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/Admin/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">

</head>

<body link="#0563C1" vlink="#954F72">

Consumer | Indexer | End point | Response Time
-- | -- | -- | --
London | Europe | https://gsql-proxy | 500 ms
London | Europe | subquery-testnet.brightlystake.com | 40 ms

</body>

</html>

However when the test is repeated using a customer location from Chicago the response times are more or less equal. A quick check seems to indicate that https://gsql-proxy endpoint servers are both located in US-EAST location which maybe causing the additional delay due to round-trips from London to Us adding to the response time.

**Comment by ianhe8x** [comment]
Actually the requests are not routed to the proxy. The load generator doesn't have limitation of using http endpoints.

**Comment by ianhe8x** [comment]
But I did see an indexer who put gql-proxy as their indexer url in the indexer metadata, which I think shouldn't be encouraged. gql-proxy is a fall-back solution only for indexer doesn't have https certs.

**Reply by Staking7pc** [reply]
How exactly do we update  in indexer metadata ? The default query URL is visible with gsql-proxy when checked in the projects-> indexer tab 
![image](https://user-images.githubusercontent.com/81879738/169045732-5c08d286-8cdd-47db-97ab-98c53a069b19.png)

**Reply by ianhe8x** [reply]
In the UI, we added proxy for indexers whose endpoint isn't http.
The place for indexer to specify their endpoint is in the indexer app.
<img width="929" alt="image" src="https://user-images.githubusercontent.com/39037239/169162932-2878cd0a-748d-4c10-b5b4-5d8f16979b08.png">

**Reply by Staking7pc** [reply]
Hi Ian 
I have already set the end point in the admin page.
Even for your indexers I can see only the gsql-proxy endpoint in the indexers page(https://frontier.subquery.network/explorer/)  of the projects . 

**Where are the consumers supposed to get the end point for querying from their indexers ?**

I think it would be better to add a query URL in the (https://frontier.subquery.network/plans/service-agreements/ongoing) so that they can get the query URL of the indexers with who they have the service agreement instead of going through the entire list of indexers. This page can have the direct endpoint without the gsql proxy 

![image](https://user-images.githubusercontent.com/81879738/169176684-f6b53a60-e77d-41f0-9dbf-2ea4be1cb4f9.png)

**Reply by ianhe8x** [reply]
I got your point, we can add a button so consumer can copy the original url.
For your suggestion, we are adding playground for ongoing agreements, you will see it soon within this week.
